asllib/Typing.ml

```diff
@@ -1961,6 +1977,7 @@ module Annotate (C : ANNOTATE_CONFIG) = struct
         let new_env, new_ldis =
           List.fold_right2
             (fun ty' ldi' (env', les) ->
+              (* TODO: propagate epxressions? *)
```

Comment:
I think that if you perform this inside the static environment then the `annotate_local_decl_item` below would do the right thing.

Reply:
This is related to the other comment about making sure that expressions always refer to the earliest variables. I thought it would be good to have just one function where immutable expressions are added to the environment and replacement of variables by their equivalent expressions happens. This ensures that if you always use this function the invariant of referring to the oldest expressions holds.